m4/ax_check_compile_flag.m4

```diff
@@ -41,7 +41,10 @@ AC_DEFUN([AX_CHECK_COMPILE_FLAG],
 AS_VAR_PUSHDEF([CACHEVAR],[ax_cv_check_[]_AC_LANG_ABBREV[]flags_$4_$1])dnl
 AC_CACHE_CHECK([whether the _AC_LANG compiler accepts $1], CACHEVAR, [
   ax_check_save_flags=$[]_AC_LANG_PREFIX[]FLAGS
-  _AC_LANG_PREFIX[]FLAGS="$[]_AC_LANG_PREFIX[]FLAGS $4 $1 -Werror"
+  if test x"$GCC" = xyes -o x"$GFC" = xyes -o x"$G77" = xyes -o x"$GOBJC" = xyes -o x"$GOBJCXX" = xyes ; then
+    add_gnu_werror="-Werror"
+  fi
+  _AC_LANG_PREFIX[]FLAGS="$[]_AC_LANG_PREFIX[]FLAGS $4 $1 $add_gnu_werror"
```

Comment:
You're missing `$GXX` for C++.

Also the autoconf manual advises to avoid `test -o` and instead use `||`, i.e. `test x"$GCC" = xyes ||test x"$GFC" = xyes` etc.

However you'll add `-Werror` if **any** of the probed compilers is detected as the GNU one, so if you're using a GNU compiler for one language and a non-GNU one for another, both will get `-Werror` but the non-GNU one may not support it.  That's certainly a corner case, but these macros aim to be reusable so should try to cope with unusual situations.

I think it should just check the variable for the current language - here's an implementation which picks the appropriate variable to check at m4 time (with a fall-back to not adding `-Werror` for languages where there's no GNU compiler):

```
if test x"m4_case(_AC_LANG,
                   [C], [$GCC],
                   [C++], [$GXX],
                   [Fortran], [$GFC],
                   [Fortran 77], [$G77],
                   [Objective C], [$GOBJC],
                   [Objective C++], [$GOBJCXX],
                   [no])" = xyes ; then
  add gnu_werror="-Werror"
fi
```

(Lightly tested - I manually checked the generated code for C, C++ and the fall-back case.)

Reply:
Thank you for your educative comments.

> You're missing $GXX for C++.

Good point, not sure how that happened, I looked through the manual and maybe there was no comment about the variable... or I just missed it.

> Also the autoconf manual advises to avoid `test -o`...

Well, it's all over the place in the archive, I just looked at another macro and copied & pasted from there. Maybe it makes sense to do a mass replacement, otherwise it gets really confusing.

> (Lightly tested - I manually checked the generated code for C, C++ and the fall-back case.)

Works for me, committed a fixup.